### ISSUE TITLE
Some UnionType fragments are not triggering the correct loading state for additional queries. 

### ISSUE THREAD

**Issue by acorcutt** [apollographql/apollo-client#857]
Its only showing a loading state when fetching the 1st fragment from a union, is this related to what is being fixed in #707 or is the loading state looking at the wrong fragment. Data is loaded correctly for all types and re-ordering changes things...

Site is a union...

```
export default new GraphQLUnionType({
  name: 'Site',
  types: () => ([TeamType, MemberType])
});
```

1st query triggers loading state as expected for either type...

```
graphql(gql`
  query ParentQuery {
    site {
      __typename
      ... on Team {        
        id
        name
      }    
      ... on Member {        
        id
        name
      }    
    }
  }`)(Parent);

```

2nd query from child of root triggers loading state when returned type is a Team but not Member...

```
graphql(gql`
  query ChildQuery {
    site {
      __typename
      ... on Team {        
        id
        name
        summary
      }    
      ... on Member {        
        id
        name
        summary
      }    
    }
  }`)(Child);
```

Re-ordering the fragments triggers loading state when type is a Member but not a Team...

```
graphql(gql`
  query ChildQuery {
    site {
      __typename
      ... on Member {        
        id
        name
        summary
      }    
      ... on Team {        
        id
        name
        summary
      }    
    }
  }`)(Child);
```

**Comment by helfer** [comment]
@acorcutt Thanks for bringing this up. Are you using version 0.5.0 of Apollo Client? And is `__typename` added to the query (i.e. do you have `addTypename` set to true (or at least not to false))?

FWIW I don't think this is related to #707. It seems to be related to fragment handling instead.

**Comment by acorcutt** [comment]
I've tried both with & without addTypename and my id's are globally unique from the relay node interface, versions: apollo-client@0.5.0 & react-apollo@0.5.16

I'm thinking it is the fragment matching now, I'm not seeing it batch any queries with fragments -  #771 suggests there is no matching on unions?

**Comment by helfer** [comment]
As long as there is no data there initially, Apollo Client should show the loading state regardless of what type the returned object is going to be. And given __typename, I see no reason why Apollo Client wouldn't be able to figure out which fragment applies.

@stubailo what's the status on supporting this simple case of union types from the example above in graphql-anywhere.

**Comment by helfer** [comment]
@acorcutt is this still an issue with the latest version?

**Comment by helfer** [comment]
This should be fixed with #1483.
